### ISSUE TITLE
[iOS] Fairplay with different key per asset not playing

### ISSUE THREAD

**Issue by facugu1998** [TheWidlarzGroup/react-native-video#3227]
# Bug

## Platform
Which player are you experiencing the problem on:
* iOS

## Description
When a content has mutliple asset and each asset has different key, for example, skd://{key}/AUDIO for audio playlist, skd://{key}/HD and skd://{key}/SD for video playlist, the content is not being played in most cases and it throws .

```
 {"error": {"code": -19152, "domain": "CoreMediaErrorDomain", "localizedDescription": "The operation couldn’t be completed. (CoreMediaErrorDomain error -19152 - No se pudo completar la operación. (Error de CoreMediaErrorDomain -19152))", "localizedFailureReason": "", "localizedRecoverySuggestion": ""}, "target": 143}
``` 

This issue is related to #2881 but in version 6.0.

As prerequisite you should change:
`let contentId = _drm.contentId ?? loadingRequest.request.url?.host`
to
`let contentId = _drm.contentId ?? loadingRequest.request.url?.absoluteString`

in RCTResourceLoaderDelegate.swift in order to send the complete key string to the license server, otherwise the licenseServer doesn't recognize the sent key.

## Environment info

React native info output:

System:
    OS: macOS 12.5.1
    CPU: (8) arm64 Apple M2
    Memory: 88.39 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 20.5.1 - /opt/homebrew/bin/node
    Yarn: Not Found
    npm: 9.8.0 - /opt/homebrew/bin/npm
    Watchman: 2023.08.28.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.12.1 - /Users/ingenieria/.rvm/gems/ruby-2.7.5/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK:
      API Levels: 29, 30, 31, 33, 33
      Build Tools: 29.0.2, 30.0.3, 31.0.0, 33.0.0, 33.0.1
      System Images: android-31 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2022.1 AI-221.6008.13.2211.9477386
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.18 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0 
    react-native: 0.71.3 => 0.71.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

Library version: 6.0.0-alpha.7

## Steps To Reproduce

1. Select FairPlay content with different key per asset.
2. The player throws: "The operation couldn't be completed" and the content does't play.

## Expected behaviour

1. The player should play.

## Reproducible sample code

The sample code should work, but the getLicense function must be the following:

```
import { Base64 } from "js-base64";
getLicense: async (spc, skd) => {
          return fetch(licenseServer, {
            method: "POST",
            headers: {
              "Content-Type": "application/octet-stream",
              keyId: skd,
            },
            body: Base64.toUint8Array(spc),
          })
            .then((response) => response.blob())
            .then(async (response) => {
              const base_64 = await blobToBase64(response);
              return base_64.split(",")[1];
            })
            .catch((error) => {
              console.error("Error", error);
            });
        },
      },
```

## Video sample

URL: https://cdnw1-stg.zetatv.com.uy/testzapp/bigbuckbunny/playlist.m3u8
licenseServer: https://pm-stg.zetatv.com.uy:8443/api/fairplay/get_license/fbdd541b-0036-4b4c-afef-351b7627ba81/
certificateUrl: https://pm-stg.zetatv.com.uy:8443/fairplay/cert.crt

**Comment by freeboub** [comment]
Hi, I am sorry I didn't have time to continue investigations, but my pr should be a good starting point ?

**Comment by facugu1998** [comment]
Hi @freeboub, we tried the patch you proposed on #2881 and it didn't work for us. Were you able to make a PR with more changes based on it?

**Comment by freeboub** [comment]
linked PR has been merged in 6.0.0-alpha.8, let's close this issue
